dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt

```diff
@@ -272,6 +272,7 @@ internal constructor(
             span.context(),
             Format.Builtin.TEXT_MAP_INJECT,
             TextMapInject { key, value ->
+                tracedRequestBuilder.removeHeader(key)
```

Comment:
maybe we can add a comment here otherwise I am pretty sure we will forget why we are doing this.